### PR TITLE
[MIRROR] Merges AI Branch into Master

### PR DESCRIPTION
**Pull request by VirgoBot**
Original PR: https://github.com/PolarisSS13/Polaris/pull/5735
--------------------
Something is probably going to conflict 40+ times over with the same files again before the master sync finishes so might be best to shove this here now with the sync included.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
So many conflicts...

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -108,18 +108,25 @@
 	light_range=5
 	light_color="#ff0000"
 	spawnable=list(
+<<<<<<< HEAD
```
Comment (Cameron653) [inline code/game/gamemodes/cult/cult_structures.dm]: Just for reference, there's a conflict here.

File: code/game/gamemodes/technomancer/spells/control.dm [hunk 2]
```diff
@@ -21,6 +21,7 @@
 	aspect = ASPECT_BIOMED //Not sure if this should be something else.
 	var/image/control_overlay = null
 	var/list/controlled_mobs = list()
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/game/gamemodes/technomancer/spells/control.dm]: Also here.

File: code/game/gamemodes/technomancer/spells/summon/summon_creature.dm [hunk 3]
```diff
@@ -16,6 +16,7 @@
 	desc = "Chitter chitter."
 	summoned_mob_type = null
 	summon_options = list(
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/game/gamemodes/technomancer/spells/summon/summon_creature.dm]: And here

File: code/game/objects/items/weapons/tools/weldingtool.dm [hunk 4]
```diff
@@ -1,3 +1,7 @@
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/game/objects/items/weapons/tools/weldingtool.dm]: Also here

File: code/game/objects/items/weapons/tools/weldingtool.dm [hunk 5]
```diff
@@ -317,7 +321,10 @@
 		var/obj/item/organ/internal/eyes/E = H.internal_organs_by_name[O_EYES]
 		if(!E)
 			return
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/game/objects/items/weapons/tools/weldingtool.dm]: Here too

File: code/modules/mob/living/death.dm [hunk 6]
```diff
@@ -1,12 +1,18 @@
 /mob/living/death()
 	clear_fullscreens()
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/modules/mob/living/death.dm]: Actually they're basically fucking everywhere. <_>

**Comment by SpadesNeil** [comment]
I have managed to remove all conflicts. Don't expect anything to work correctly, though. Or even compile.

I am also 98% sure I broke whatever our vorestation edit did in death.dm, but this has been sitting here so long and backlogging Github that I don't care. Let shit break. Maybe everything being broken will motivate someone who knows better to _fix_ it.

File: vorestation.dme [hunk 7]
```diff
@@ -2223,6 +2238,7 @@
 #include "code\modules\mob\living\silicon\robot\subtypes\lost_drone.dm"
 #include "code\modules\mob\living\silicon\robot\subtypes\syndicate.dm"
 #include "code\modules\mob\living\simple_animal\corpse.dm"
+<<<<<<< HEAD:vorestation.dme
```
Comment (Cameron653) [inline vorestation.dme]: Last bit which needs to be changed

File: vorestation.dme [hunk 8]
```diff
@@ -2315,6 +2331,103 @@
 #include "code\modules\mob\living\simple_animal\vore\shadekin\shadekin.dm"
 #include "code\modules\mob\living\simple_animal\vore\shadekin\types.dm"
 #include "code\modules\mob\living\simple_animal\vore\shadekin\~defines.dm"
+=======
```
Comment (Cameron653) [inline vorestation.dme]: And this

File: vorestation.dme [hunk 9]
```diff
@@ -2315,6 +2331,103 @@
...
+#include "code\modules\mob\living\simple_mob\subtypes\plant\tree.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\slime.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\feral\feral.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\combat.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\consumption.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\defense.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\discipline.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\subtypes.dm"
+#include "code\modules\mob\living\simple_mob\subtypes\slime\xenobio\xenobio.dm"
+>>>>>>> 3155d58... Merge pull request #5735 from Neerti/hopefully_last_master_sync:polaris.dme
```
Comment (Cameron653) [inline vorestation.dme]: And this here

**Comment by SpadesNeil** [comment]
I didn't touch the .dme on purpose; that can be fixed last.

**Comment by Cameron653** [comment]
Also, I'd like to add a note that everything has been changed to "simple_mob" instead of "simple_animal" which means that we'll need to change that on our mobs later.

**Comment by SpadesNeil** [comment]
I have reduced the compile errors to 100 errors, all of which are the result of now undefined mobs.

This should be easy to fix from here for those of you who know what you are doing with mobs, which I do not. We need to port our mob vore system to the new mobs. As I said before, going forward, we will not use separate mobs from the vanilla mobs. All mobs that can do vore will only be the one mob.

**Comment by Cameron653** [comment]
I believe the quickest way to solve a mass majority of the errors would be to change all mentions of "simple_animal" to "simple_mob". Doing such will probably still leave a few errors, but wipe out a mass majority of them.

File: code/modules/mob/living/death.dm [hunk 10]
```diff
@@ -1,18 +1,7 @@
 /mob/living/death()
 	clear_fullscreens()
```
[the remark this answers is not in the document]
Comment (SpadesNeil) [reply code/modules/mob/living/death.dm]: Post again but with ``` instead of `

[the remark this answers is not in the document]
Comment (Novacat) [reply code/modules/mob/living/death.dm]: Done. Thank you.

**Comment by SpadesNeil** [comment]
That caused a conflict I can't fix in here.

**Comment by Poojawa** [comment]
Like I said in the discord, the only checks failing at this point are the Unit checks because someone had an active turf on z5 near engineering.

Everything else that isn't direct NPC conversion is resolved, so all of the submaps should work as intended now for testing.

**Comment by SpadesNeil** [comment]
The hell is there an active edge for?

**Comment by Poojawa** [comment]
There's also like, 50+ random pipes that aren't connected (though they might be the up/down Z pipes, idk) and a bunch of random power cables not connected. Tether needs someone to fine-comb all the little things.

**Comment by Sebbe9123** [comment]
I believe some of that might be the construction sites.

**Comment by SpadesNeil** [comment]
If I could get ten goddamn minutes to myself, I could probably fix this. It doesn't seem like it's that complicated of an issue. Active edges are an easy fix.

**Comment by SpadesNeil** [comment]
https://www.youtube.com/watch?v=QJFM86tgVxY

**Comment by Novacat** [comment]
Active edge is related to expedition areas, so, something the fix did not touch unfortunately. There is an old issue where expedition areas would rarely fail travis with active edges, so you can probably fix this by just having Travis do another run.

**Comment by Poojawa** [comment]
it doesn't fix the problem, it's more likely the unit test run happens to spawn it.

**Comment by Novacat** [comment]
Unless this is something new, which I highly doubt, it is an old active edge error that has been on Virgo's codebase for a long time. It shows up extremely rarely in expedition areas, and is not really reproducible to my knowledge. You can probably rerun Travis again and it will pass.

**Comment by SpadesNeil** [comment]
I have rerun Travis three times now.

**Comment by Novacat** [comment]
https://github.com/VOREStation/VOREStation/pull/4774 Is a possible fix.

**Comment by SpadesNeil** [comment]
nope, but it fixed the map diff renders

**Comment by Heroman3003** [comment]
According to checker, the Travis ran fine though?

**Comment by SpadesNeil** [comment]
-heavy breathing- 👀

**Comment by Novacat** [comment]
I would recommend adding another bullet point, however, this PR still has a number of runtimes that need to be fixed.

**Comment by Heroman3003** [comment]
Also, pretty sure not all vore mobs have been fully converted yet, even without vore capabilities in tow.

**Comment by Novacat** [comment]
https://github.com/VOREStation/VOREStation/pull/4709

The Runtime fix was included in here because I used cat_vr to debug the cause of the runtime.

**Comment by Screemonster** [comment]
HOLY SHIT DID TRAVIS PASS

**Comment by Novacat** [comment]
Runtimes are gone. Travis is passing. There are non-runtime errors with the spawners still, but that is not a big thing (and is fixed in Sebbe's PR anyway). Fuck you mapbot.

Currently working on attempting to port Shadekin. I defiantly will not be able to port their AI but they are supposed to be player-controlled 99% of the time anyway.

**Comment by SpadesNeil** [comment]
image
https://www.youtube.com/watch?v=b-LTmNmNPO0

**Comment by Novacat** [comment]
WHERE DID THAT ONE COME FROM!?

No worries, I will figure it out.

**Comment by SpadesNeil** [comment]
At this point you'd just have to fix the mobs_to_pick_from lists and we're good. No shadekin on it though. I know the old list had shadekin but let's not on this new list.

**Comment by Novacat** [comment]
That should be handled by https://github.com/VOREStation/VOREStation/pull/4790 which also fixes a bunch of other stuff too.

**Comment by Novacat** [comment]
Mapbot shit itself again. Travis is passing, the one singular mobs_to_pick_from error is from a shadekin spawner, so that can be ignored.

**Comment by SpadesNeil** [comment]
There's some other shit that should be fixed here but this looks like it's okay to merge.

Fingers crossed.

**Pull request merged**